src/platform/silabs/tracing/SilabsTracing.h

```diff
@@ -0,0 +1,205 @@
+/*
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <cstddef>
+#include <cstdlib>
+#include <lib/core/CHIPError.h>
+#include <lib/core/CHIPPersistentStorageDelegate.h>
+#include <lib/support/LinkedList.h>
+#include <stdint.h>
+#include <system/SystemClock.h>
+
+#ifndef SERIALIZED_TIME_TRACKERS_SIZE_BYTES
+// Default size, watermarks store 6 uint32_t, which is 24 bytes
+// We currently have 19 operations to track, so 19 * 24 = 456 bytes
+// 512 bytes should be enough including the serialization overhead
+#define SERIALIZED_TIME_TRACKERS_SIZE_BYTES 512
+#endif
+
+namespace chip {
+namespace Tracing {
+namespace Silabs {
+
+// Enum for the different operation to trace
+enum class TimeTraceOperation : uint8_t
+{
+    kSpake2p,
+    kPake1,
+    kPake2,
+    kPake3,
+    kOperationalCredentials,
+    kAttestationVerification,
+    kCSR,
+    kNOC,
+    kTransportLayer,
+    kTransportSetup,
+    kFindOperational,
+    kCaseSession,
+    kSigma1,
+    kSigma2,
+    kSigma3,
+    kOTA,
+    kImageUpload,
+    kImageVerification,
+    kAppApplyTime,
+    kBootup,
+    kSilabsInit,
+    kMatterInit,
+    kNumTraces,
+};
+
+enum class OperationType : uint8_t
+{
+    kBegin,
+    kEnd,
+    kInstant,
+};
+
+struct TimeTracker
+{
+    // Temporary values
+    System::Clock::Milliseconds32 mStartTime;
+    System::Clock::Milliseconds32 mEndTime;
+    TimeTraceOperation mOperation;
+    OperationType mType;
+    CHIP_ERROR mError;
+};
+
+struct Watermark
+{
+    // Values that will be stored in the NVM
+    System::Clock::Milliseconds32 mMovingAverage; // Successful operation average time
+    System::Clock::Milliseconds32 mMaxTimeMs;     // Successful operation max time
+    System::Clock::Milliseconds32 mMinTimeMs;     // Successful operation min time
+    uint32_t mTotalCount;                         // Total number of times the operation was initiated
+    uint32_t mSuccessfullCount;                   // Number of times the operation was completed without error
+    uint32_t mCountAboveAvg;                      // Number of times the operation was above the average time
+};
```

Comment:
If these are just used by the class, we should move them within the class and in the private section if necessary

Reply:
I'm planning to have access to them in the tests at least, and maybe I'll need it not to be private so I'll leave that here for now, might move it in further PR.